pkg/core/interop/context.go

```diff
@@ -113,7 +114,7 @@ func NewContractMD(name string) *ContractMD {
 	emit.Syscall(w.BinWriter, interopnames.NeoNativeCall)
 
 	c.Script = w.Bytes()
-	c.Hash = hash.Hash160(c.Script)
+	c.Hash = state.CreateContractHash(hash.Hash160([]byte{byte(opcode.PUSH1)}), c.Script)
```

Comment:
We better merge #1610 now first and then this one. And add some test that specifically checks for native contract's hashes and compares them to known values from C# devpack (see neo-project/neo-devpack-dotnet#391).

Reply:
We can also add some method like `CreateNativeContractHash(name string)` to reduce code duplication between native contracts.

Reply:
Took it directly from C# node, devpack has outdated values. Fixed.